src/opera_disp_tms/generate_sw_disp_tile.py

```diff
@@ -117,18 +117,23 @@ def add_granule_data_to_array(
     Returns:
         The updated short wavelength cumulative displacement array and the secondary date of the granule
     """
-    granule_dataarray = open_opera_disp_granule(granule.s3_uri, 'short_wavelength_displacement')
+    datasets = ['short_wavelength_displacement', 'connected_component_labels']
+    granule_xr = open_opera_disp_granule(granule.s3_uri, datasets)
 
-    same_ref_date = round_to_day(granule_dataarray.attrs['reference_date']) == round_to_day(frame.reference_date)
+    same_ref_date = round_to_day(granule_xr.attrs['reference_date']) == round_to_day(frame.reference_date)
     if not same_ref_date:
         raise NotImplementedError('Granule reference date does not match metadata tile, this is not supported.')
 
-    granule_dataarray = granule_dataarray.rio.reproject('EPSG:3857', transform=geotransform, shape=frame_map.shape)
+    bbox = create_buffered_bbox(geotransform.to_gdal(), frame_map.shape, 120)  # EPSG:3857 is in meters
+    granule_xr = granule_xr.rio.clip_box(*bbox, crs='EPSG:3857')
```

Comment:
This buffering/clipping is purely for performance, yes? To avoid reading the full extent of the displacement and connected component layers into memory?

Reply:
Yes, it's purely for performance. If we subset to a bounding box first, we don't need to download the sw_disp/conn_comp data for the granule's entire extent.